### PR TITLE
[trustar-150]: TruSTAR v2

### PR DESCRIPTION
**Pull request by hernantain**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

fixes: https://github.com/demisto/etc/issues/24555

## Description
New Version of TruSTAR integration. 
Rewriting the code from v1, adding tests, docstrings on code functions, adding new commands. 

For commands returning indicators we are putting data on 3 contexts: 
- The standard one without the malicious field as TruSTAR doesn't have a score for every indicator at this moment.
- DBotScore context with score as 0 for the same reason.
- TruSTAR context with all the information returned by the command.

## Screenshots
Indicator outputs on DBot, Standard and TruSTAR context

476BDB0C-044D-4FB0-BC1B-86BA25E3BB69_1_105_c

## Minimum version of Demisto
- [ ] 4.5.0
- [x] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No: Is a new integration.

## Must have
- [x] Tests
- [x] Documentation 

## Demisto Partner?
- [x] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml [hunk 1]
```diff
@@ -0,0 +1,1020 @@
...
+      intelligence sources on the TruSTAR Marketplace.
+    #5
+
+  - name: trustar-get-whitelisted-indicators
+    arguments:
+    - name: max-entries
+      description: The max number of results to return. Max value possible is 1000.
+      defaultValue: "25"
+    outputs:
+    - contextPath: TruSTAR.WhitelistedIndicators
```
Comment (Itay4) [inline Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml:473]: no need for this to be here, only the full context path is needed

File: Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml [hunk 2]
```diff
@@ -0,0 +1,1020 @@
...
+    - name: tags
+      description: a list of names of tags to filter by; only reports containing ALL
+        of these tags will be returned
+      isArray: true
+    - name: excluded-tags
+      description: reports containing ANY of these tags will be excluded from the
+        results.
+      isArray: true
+    outputs:
+    - contextPath: TruSTAR.Report.reportTitle
```
Comment (Itay4) [inline Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml:557]: please add output types here

File: Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml [hunk 3]
```diff
@@ -0,0 +1,1020 @@
...
+      - CVE
+      - REGISTRY_KEY
+      description: The type of the indicator to delete.
+    description: Delete an indicator from the user’s company whitelist.
+      #17
+
+  - name: trustar-get-phishing-submissions
+    arguments:
+    - name: priority_event_score
+      description: Score of email submission
```
Comment (Itay4) [inline Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml:861]: what is the optional range here ? -1 to 3? please mention it in the description

File: Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml [hunk 4]
```diff
@@ -0,0 +1,1020 @@
...
+    description: Marks a phishing email submission with one of the phishing namespace
+      tags
+    #19
+
+  - name: trustar-get-phishing-indicators
+    arguments:
+    - name: normalized_indicator_score
+      description: Intel scores to restrict the query.
+      isArray: true
+      defaultValue: -1,0,1,2,3
```
Comment (Itay4) [inline Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml:928]: what is the optional range here ? -1 to 3? please mention it in the description

File: Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml [hunk 5]
```diff
@@ -0,0 +1,1020 @@
...
+    - name: priority_event_score
+      description: Score of email submission to restrict the que
```
Comment (Itay4) [inline Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml:930]: what is the optional range here ? -1 to 3? please mention it in the description

**Comment by hernantain** [comment]
@Itay4 I've:

- Corrected all small changes, typos, descriptions, parameters, etc.
- Moved Integration to existring TruSTAR pack
- Deprecating v1
- Implemented test-module
- Added test playbook

About the inconsistency with correlation_count on the different commands, we'll be solving the bug from our side. Same with domain/URL types. The `search-indicators` and `indicators-metadata` return the same TS object but the application differs within those 2, actually they can easily be linked on a playbook. And last, we are scoring DBot Indicators as Unknown as most of our indicators do not inform a malicious score. We prefer to change that once we can do it the same way for all of them.

File: Packs/TruSTAR/Integrations/TruSTAR_V2/TruSTAR_V2.py [hunk 6]
```diff
@@ -0,0 +1,937 @@
...
+        }
+        return d
+
+    def get_report_deep_link(self, report_id):
+        """ Returns report link on TruSTAR self.station. """
+        return f'{self.station}/constellation/reports/{report_id}'
+
+
+    def test_module(self):
+        response = self.client.ping()
```
[the remark this answers is not in the document]
Comment (hernantain) [reply Packs/TruSTAR/Integrations/TruSTAR_V2/TruSTAR_V2.py:341]: It justs checks connectivity, the API does not verify the api key but of course that if the credentials are not OK, this command is going to fail.

**Comment by Itay4** [comment]
please revert the removal of https://github.com/demisto/content/blob/360918b7019aa5018b8478053883630600331f57/Pipfile

File: Packs/TruSTAR/Integrations/TruSTAR_V2/README.md [hunk 7]
```diff
@@ -0,0 +1,574 @@
+Use the TruSTAR v2 integration to manage reports, indicators and phishing submissions.
```
Comment (Itay4) [inline Packs/TruSTAR/Integrations/TruSTAR_V2/README.md:1]: The README seem to be copy and pasted from the previous integration, and it is not according to our standard
please regenerate as described in https://xsoar.pan.dev/docs/integrations/integration-docs

**Comment by Itay4** [comment]
is this the expected behavior?
image

File: Packs/TruSTAR/Integrations/TruSTAR_V2/README.md [hunk 8]
```diff
@@ -0,0 +1,839 @@
+TruSTAR is an Intelligence Management Platform that helps you operationalize data across tools and teams, helping you prioritize investigations and accelerate incident response.
+This integration was integrated and tested with version xx of TruSTAR v2
```
Comment (Itay4) [inline Packs/TruSTAR/Integrations/TruSTAR_V2/README.md:2]: can we set the `xx` version to something else?

File: Packs/TruSTAR/Integrations/TruSTAR_V2/README.md [hunk 9]
```diff
@@ -0,0 +1,839 @@
...
+| Account.Email.Address | String | The email address of the account. | 
+| RegistryKey.Path | String | The path to the registry key | 
+| Domain.Name | String | The domain name, for example: "google.com". | 
+| DBotScore.Indicator | string | The indicator we tested | 
+| DBotScore.Type | string | The type of the indicator | 
+| DBotScore.Vendor | string | Vendor used to calculate the score | 
+| DBotScore.Score | number | The actual score | 
+
+
+#### Command Example
```
Comment (Itay4) [inline Packs/TruSTAR/Integrations/TruSTAR_V2/README.md:61]: please add examples for all commands of command executions and outputs

**Pull request merged**